### PR TITLE
✨ hack/observability: Add Parca, Bump observability tools, Add Grafana scale dashboard

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: hack/observability/grafana/kustomization.yaml [hunk 1]
```diff
@@ -1,10 +1,15 @@
 resources:
   - ../namespace.yaml
+  # The kube-state-metrics helm chart will reference a configmap with name `kube-state-metrics-crd-config`.
+  # The configMapGenerator below will create the configmap and append a hash suffix calculated from its
+  # content to the name. Kustomize will append the suffix hash to all references in the helm chart, but
+  # only when the helm chart content is referenced in "resources".
+  # This would not work if the helm chart is configured in this file via the "helmCharts" option.
```
Comment (fabriziopandini) [inline hack/observability/grafana/kustomization.yaml:7]: This is a little bit confusing, why we need to take care of creating a config map for kube state metrics while installing grafana?

**Review by fabriziopandini — COMMENTED** [review]
Grat job in updating all the charts and kudos for kicking of the work about creating grafana dashboard for our test environment

Q: Might be I'm wrong, but I don't see the part installing Parca

File: hack/observability/grafana/kustomization.yaml (hunk 1 above)
Comment (sbueringer) [reply hack/observability/grafana/kustomization.yaml]: Should be fixed now. This should be:
```
  # The grafana helm chart will reference a configmap with name `grafana-dashboards`.
```

Already had it locally but didn't push the last commit

**Comment by sbueringer** [comment]
> Q: Might be I'm wrong, but I don't see the part installing Parca

Thx good catch. Forgot to push the last commit :/. Should be there now

File: hack/tools/tilt-prepare/main.go [hunk 2]
```diff
@@ -841,6 +841,14 @@ func prepareWorkload(name, prefix, binaryName, containerName string, objs []unst
 				}
 				args = debugArgs
 
+				// Set annotation to point parca to the profiler port.
+				if d.ProfilerPort != nil && *d.ProfilerPort > 0 {
+					if deployment.Spec.Template.Annotations == nil {
+						deployment.Spec.Template.Annotations = map[string]string{}
+					}
+					deployment.Spec.Template.Annotations["parca.dev/port"] = "6060"
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply hack/tools/tilt-prepare/main.go:849]: Nope. This should point to the port in the container which is always 6060 (see l. 832)

ProfilerPort is the one outside of kind on your local machine

**Pull request merged**